### PR TITLE
htpasswd: refactor

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY

A bit refactor on the module.

##### ISSUE TYPE

- Refactoring Pull Request

##### COMPONENT NAME

htpasswd

### CODE REVIEW

File: changelogs/fragments/6901-htpasswd-refactor.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - htpasswd - minor code improvements in the module (https://github.com/ansible-collections/community.general/pull/6901).
```
Comment (felixfontein) [inline changelogs/fragments/6901-htpasswd-refactor.yml:2]: Please add a `removed_features` entry that support for passlib < 1.6 was removed. The requirements have been passlib >= 1.6 for a long time now, but the code always supported older versions (apparently).

(This should also only be merged to main and not backported, as the actual drop of 1.6 support should happen in a new major release and not in a feature release.)

File: plugins/modules/htpasswd.py [hunk 2]
```diff
@@ -260,11 +227,12 @@ def main():
...
-            try:
-                [f.write(line) for line in lines if line.strip()]
-            finally:
-                f.close()
+            with open(path, "w") as f:
+                f.writelines(line for line in lines if line.strip())
+
+    except IOError:
+        # No preexisting file to remove blank lines from
+        f = None
```
Comment (felixfontein) [inline plugins/modules/htpasswd.py:235]:
```suggestion
        pass
```

File: plugins/modules/htpasswd.py [hunk 3]
```diff
@@ -233,20 +208,12 @@ def main():
     create = module.params['create']
     check_mode = module.check_mode
 
-    if not passlib_installed:
-        module.fail_json(msg=missing_required_lib("passlib"), exception=PASSLIB_IMP_ERR)
+    deps.validate(module)
 
     # Check file for blank lines in effort to avoid "need more than 1 value to unpack" error.
```
Comment (felixfontein) [inline plugins/modules/htpasswd.py:213]: I wonder if this 'hack' below is still needed. Maybe this was necessary for an older version of passlib.

Comment (russoz) [reply plugins/modules/htpasswd.py:213]: Yup, maybe, but I will leave that for another PR, with tests supporting the hypothesis.

File: plugins/modules/htpasswd.py (hunk 2 above)
Comment (russoz) [reply plugins/modules/htpasswd.py:235]: done

File: changelogs/fragments/6901-htpasswd-refactor.yml (hunk 1 above)
Comment (russoz) [reply changelogs/fragments/6901-htpasswd-refactor.yml:2]: done

File: plugins/modules/htpasswd.py (hunk 3 above)
Comment (felixfontein) [reply plugins/modules/htpasswd.py:214]: Yes, definitely!

**Pull request merged**